interface/ceed-operator.c

```diff
@@ -904,6 +835,101 @@ int CeedOperatorSetField(CeedOperator op, const char *field_name,
   return CEED_ERROR_SUCCESS;
 }
 
+/**
+  @brief Get the CeedOperatorFields of a CeedOperator
+
+  Note: Calling this function asserts that setup is complete
+          and sets the CeedOperator as immutable.
+
+  @param op                  CeedOperator
+  @param[out] input_fields   Variable to store input_fields
+  @param[out] output_fields  Variable to store output_fields
+
+  @return An error code: 0 - success, otherwise - failure
+
+  @ref Advanced
+**/
+int CeedOperatorGetFields(CeedOperator op, CeedInt *num_input_fields,
+                          CeedOperatorField **input_fields,
+                          CeedInt *num_output_fields,
+                          CeedOperatorField **output_fields) {
```

Comment:
@pazner, relevant to both the Julia interface and the MFEM integration, this is an API change. The old function signature was
```C
int CeedOperatorGetFields(CeedOperator op,
                          CeedOperatorField **input_fields,
                          CeedOperatorField **output_fields)
```
A similar change was made for `CeedQFunctionGetFields()`.

Reply:
You could get around it fine in MFEM with
```C
#if CEED_VERSION_GE(0, 9, 1)
#  define CeedOperatorGetFields(a,b,c) CeedOperatorGetFields(a,NULL,b,NULL,c)
#else
#  define CeedOperatorGetFields(a,b,c) CeedOperatorGetFields(a,b,c)
#endif

Reply:
Does it make sense to bump the library version since this is a breaking change?
```c++
#define CEED_VERSION_MAJOR 0
#define CEED_VERSION_MINOR 10
#define CEED_VERSION_PATCH 0
#define CEED_VERSION_RELEASE false
```

Reply:
> You could get around it fine in MFEM with

I could do something similar in Julia too, but maybe it's better to just adopt the new interface entirely (and require a more up-to-date version of libCEED)?

Reply:
I think we're planning on a release this month, but we're a little busy here at CU the next couple of weeks though, so it would be later in the month.

Reply:
Does it make sense to bump the version already in PRs? i.e. now that 0.9.0 is released, this PR is targeting 0.10.0?

Reply:
In particular, the Julia CI is currently failing because it's attempting to use the new API with the most recent release of libCEED.

I could change the `minimum_libceed_version` in LibCEED.jl, which would then recognize that the version is incompatible and not run the tests.

Reply:
`CEED_VERSION_GE` compares as +infinity when `CEED_VERSION_RELEASE` is `false`. The next release will be 0.10. I think we can have the same logic is the Julia code; cf. #792

Reply:
That sound reasonable. To make sure I understand what you are suggesting, I can set `minimum_libceed_version` to 0.10.0, but `libceed_version >= minimum_libceed_version` will evaluate to `true` whenever `CEED_VERSION_RELEASE` is false (even if the version number is 0.9.0)?

Reply:
Thanks for those commits. With them, it looks like we're down to just the OCCA bug as a CI failure